### DISCUSSION TITLE
Append single value to time series

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#18390] [Q&A]
I'm trying to use time series to save four different values. The four different value are scores from tests that I run in a different system. When one test is done, I get a webhook message with _one_ test result of a single test, lets say the second value in the time series. Now, I tried to do something like this:
```c#
double[] values = new double[4];

values[request.Data.Type] = request.Data.Rating;

tsf.Append(request.Data.TestDate, values, TestTypeCategories.GetCategory((TestType)request.Data.Type));
```
But then the other three tests are saved as a 0 score.

Is it possible to just update a single value and keep the other three tests untouched so to speak?

**Comment by ayende** [comment]
You _can_ read the existing value, overwrite one and then save them all back. That would update this value.

Honestly, I would go with separate time series if they aren't all coming together.

**Reply by marcuslindblom** [reply]
Ok, even when having 25 tests? It should be possible to render all tests in one graph in my user interface

**Reply by ayende** [reply]
You would gather them from all the time series, and then render them, yes.
That would be the backend work of merging the result of the query.
